### ISSUE TITLE
navigator.xr.isSessionSupported extension

### ISSUE THREAD

**Issue by PlayerXYT** [immersive-web/proposals#89]
When calling `navigator.xr.isSessionSupported` you can optionally pass features as part of the `options` optional parameter. [See](https://developer.mozilla.org/en-US/docs/Web/API/XRSystem/requestSession#options).

Some optional parameters such as [hand tracking](https://developer.mozilla.org/en-US/docs/Web/API/XRSystem/requestSession#hand-tracking) add new methods to the interface, in this case `navigator.xr.hand`. Currently there is no way to test if the current device supports various features without running a null test, which is an antipattern.

```js
navigator.xr.isSessionSupported("immersive-vr"); // This example assumes the answer here is true, this should not be done normally
navigator.xr.requestSession("immsersive-vr", { optionalFeatures: [ "hand-tracking" ] });
// This is bad
const has_hand_tracking = typeof navigator.xr.hand !== "undefined" || !navigator.xr.hand;
```

What should be supported is either `options` as a part of [isSessionSupported](https://developer.mozilla.org/en-US/docs/Web/API/XRSystem/isSessionSupported#parameters) or a new function `isFeatureSupported(feature: string[]): boolean`.

```js
navigator.xr.isSessionSupported("immsersive-vr");
navigator.xr.requestSession("immsersive-vr", { optionalFeatures: [ "hand-tracking" ] });

// Option 1
const has_hand_tracking = navigator.xr.isSessionSupported("immsersive-vr", [ "hand-tracking" ]);

// Option 2
const has_hand_tracking = navigator.xr.isFeatureSupported([ "hand-tracking" ]);
```

This allows developers to create apps that support features that are not available on all headsets/controllers. In this example the developer could render fully tracked hands if the feature is available or hands in preset poses based on the triggers squeezed.

```js
const has_hand_tracking = navigator.xr.isFeatureSupported([ "hand-tracking" ]);

function render_hand() {
  if(has_hand_tracking) {
    // ...
  } else {
    switch(triggers) {
      // ...
    }
  }
}
```

**Comment by cabanier** [comment]
Yes, that proposal would make it a lot nicer to detect features. However, this will reveal too much information about the system without the need to a user action. This makes it too dangerous to expose on the web.

**Comment by PlayerXYT** [comment]
> Yes, that proposal would make it a lot nicer to detect features. However, this will reveal too much information about the system without the need to a user action. This makes it too dangerous to expose on the web.

It could be done such that `isFeatureSupported` can only be called after `isSessionSupported` just like `requestSession` does already. This allows the user a chance to block access to XR devices before any other information about the device is given.

**Comment by cabanier** [comment]
> It could be done such that `isFeatureSupported` can only be called after `isSessionSupported` just like `requestSession` does already. This allows the user a chance to block access to XR devices before any other information about the device is given.

`requestSession` can only be called during a user action. [isSessionSupported](https://immersive-web.github.io/webxr/#dom-xrsystem-issessionsupported) can be called any time.
